### ISSUE TITLE
Avoid creating a source tarball

### ISSUE THREAD

**Issue by berlincount** [genereese/togo#15]
When working with large data amounts, the files are copied around several times without good reason, costing 4x the disk space than used by the files alone in the process.

Cheers :)

**Comment by genereese** [comment]
Yes; an unfortunate side effect of the RPM building process. By Red Hat's standards, RPMs should never be excessively large in size.

You'll notice that for a lot of packages, they will be broken up into smaller pieces. For example, there is an open source game called "Wesnoth". It breaks up its files into several, smaller packages:

wesnoth
wesnoth-data
wesnoth-tools
wesnoth-server

Are you able to break up your application into smaller pieces?

**Comment by berlincount** [comment]
sorry for not getting back earlier, email overload. nope, no chance to get smaller.

looking at %setup's -T option as documented on http://www.rpm.org/max-rpm/s1-rpm-inside-macros.html, this might be what togo should do - don't create a tarball, don't unpack that tarball, hardlink the original source into the builddir during building, problem solved :)

**Comment by genereese** [comment]
That's not a terrible idea in itself, but there are several issues with directories, etc. that can not be handled by hardlinks.

Additionally, most RPMs are not built that way because the source is usually archived into a single file for ease of transport.

I'll look at adding this as a feature, however, if I can make it work.

**Comment by liedekef** [comment]
If this is still valid for some reason: cp does support a recursive copy using hardlinks while taking directories into account too (if it helps).
